docs/knowledgebase/runtime/macros.md

```diff
@@ -1,50 +1,88 @@
 
 ## Substrate Runtime Macros
 
-When developing Substrate runtime, there are key macros the majority of developers utilize. The
-following is a comprehensive explanation of these macros. After this article, a developer will have
-a better understanding of how they are used, what they do and how the expanded code is within
-[Substrate Node Template](https://github.com/substrate-developer-hub/substrate-node-template).
+The following section is a comprehensive explanation of the macros that runtime engineers most frequently encounter. 
 Developers who want to know about the implementation details are encouraged to follow the links in
-_Doc. and Example_ to see how macro code is expanded.
+"Docs and Notes" sub-sections to better understand how each macro expands.
+
+Substrate Primitives and FRAME both rely on a collection of various types of macros. The following sections will go over each in more detail. 
+Here's a general overview of Substrate macros:
+
+**Macros in the FRAME [Support Library](./docs/en/knowledgebase/runtime/frame#support-library):**
+- in `frame_support`: 
+  - [function-like](https://substrate.dev/rustdocs/v3.0.0/frame_support/index.html#macros) macros 
+  - [derive](https://substrate.dev/rustdocs/v3.0.0/frame_support/index.html#derives) macros 
+  - [attribute](https://substrate.dev/rustdocs/v3.0.0/frame_support/index.html#attributes) macros
+
+
+**Macros in the Substrate [System Library](./knowledgebase/runtime/frame#system-library):**
+- in `sp_core`: 
+  - [function-like macros](https://substrate.dev/rustdocs/v3.0.0/sp_core/index.html#macros) macros 
+  - [derive `RuntimeDebug`](https://substrate.dev/rustdocs/v3.0.0/sp_core/index.html#derives) macro
+- in `sp_runtime`: 
+  - [function-like](https://substrate.dev/rustdocs/v3.0.0/sp_runtime/index.html#macros) macros 
+  - [derive](https://substrate.dev/rustdocs/v3.0.0/sp_runtime/index.html#derives) macros
+- in `serde`: [function-like](https://substrate.dev/rustdocs/v3.0.0/sp_runtime/index.html#derives) macros
+- in `sp_api`: [function-like](https://substrate.dev/rustdocs/v3.0.0/sp_api/index.html#macros) macros
+- in `sp_std`: [function-like](https://substrate.dev/rustdocs/v3.0.0/sp_std/index.html#macros) macros
+- in `sp_version`: [function-like](https://substrate.dev/rustdocs/v3.0.0/sp_version/index.html#macros) macros
+
+> **Note**: Refer to `#Substrate dependencies` in the `Cargo.toml` file of the 
+[node template runtime](https://github.com/substrate-developer-hub/substrate-node-template/blob/master/runtime/Cargo.toml#L21) 
+to see where these are put to use. 
+
+## Overview of FRAME Versioning
+Here's a comparative overview of FRAME v1 and v2:
+- v1 relies on runtime engineers to write more code to declare traits and types in a way that is closer to learning a DSL (Domain-specific Language). The way macro inputs are structured makes it more difficult to understand what the macro is doing and how to use it. 
```

Comment:
```suggestion
- v1 relies on runtime engineers to write more code to declare traits and types in a way that is closer to learning a DSL (Domain-specific Language). The way macro inputs are structured making it more difficult to understand what the macro is doing and how to use it. 
```

Reply:
I'm not sure this would read correctly with this suggestion - we could completely rephrase and end up with:
"In v1, the input structure of macros is less clear, making it less intuitive to understand what the macros are doing and how to properly use them".

Reply:
I think original is correct too